proxy/http/HttpSM.cc

```diff
@@ -4428,24 +4428,47 @@ HttpSM::do_range_setup_if_necessary()
     do_range_parse(field);
 
     if (t_state.range_setup == HttpTransact::RANGE_REQUESTED) {
+      bool do_transform = false;
+
       if (!t_state.range_in_cache) {
         Debug("http_range", "range can't be satisfied from cache, force origin request");
```

Comment:
`t_state.num_range_fields` is `-1` here.

Reply:
Not sure what you are saying here. How can it be -1, if we have

```
t_state.range_setup == HttpTransact::RANGE_REQUESTED
```

This gets set, as far as I can tell, in HttpSM::parse_range_and_compare, no ?

Reply:
Within HttpSM::parse_range_and_compare, if

- parse failed, goto Lfaild;
- the number of valid range is not greater than 0, goto Lfaild;

At the Lfaild, `range_in_cache = false` and `num_range_fields = -1`.

`t_state.range_setup = HttpTransact::RANGE_REQUESTED` and `t_state.num_range_fields = nr` only if `nr > 0`.

The valid value of `num_range_fields` should be -1 or any number great than 0.